### PR TITLE
Features/fishing hut validator

### PR DESCRIPTION
**Pull request by Hindsight420**

### CODE REVIEW

**Comment by Hindsight420** [comment]
@JackRiga There's changes to unrelated things. See the shaders, assets, packages, unity version... Do you know what happened here?

File: Assets/_Scripts/1Data/Scriptables/BuildingType.cs [hunk 1]
```diff
@@ -1,43 +1,35 @@
...
     public GameObject Prefab;
 
     public List<ResourceValue> Cost;
 
     public bool hasRoof;
 
-    private BuildingTypeEnum BuildingTypeEnum;
-
-    public BuildingTypeEnum BuildingEnum { get => BuildingTypeEnum; }
+    public BuildingTypeEnum BuildingEnum;
```
Comment (Hindsight420) [inline Assets/_Scripts/1Data/Scriptables/BuildingType.cs:15]: Can we rename this to BuildingType?

File: Assets/_Scripts/3Managers/BaseBuildingValidator.cs [hunk 2]
```diff
@@ -1,29 +1,78 @@
...
+        var errors = new List<string>();
+        if (_tile.Building != null) { errors.Add("There's already a building there!"); }
+
+        //IsSupported
+        if (_tile.Y != 0 && map.Down(_tile.X, _tile.Y).Building is null) { errors.Add("Tile is not supported!"); }
+
+        return errors.ToArray();
+    }
+
+    //Can the given building be built on top?
```
Comment (Hindsight420) [inline Assets/_Scripts/3Managers/BaseBuildingValidator.cs:44]: Shouldn't this method be for determining whether you can build another building on top of this one?

File: Assets/_Scripts/3Managers/BaseBuildingValidator.cs [hunk 3]
```diff
@@ -1,29 +1,78 @@
...
+    //Should this building render a roof?
+    public virtual bool ShouldRenderRoof(Island map)
     {
-        return !map.GetTileAt(x, y + 1).IsOccupied;
+        return map.Up(_tile.X, _tile.Y).Building is null;
     }
 
-    public virtual string[] ValidateAdjacencies(Island map, int x, int y)
+    //Does this building care if we build it next to it's neighbours?
+    public virtual string[] ValidateAdjacencies(Island map)
```
Comment (Hindsight420) [inline Assets/_Scripts/3Managers/BaseBuildingValidator.cs]: List<string> instead? For all methods, currently it's a mix of arrays and lists.

File: Assets/_Scripts/1Data/Scriptables/BuildingType.cs (hunk 1 above)
Comment (JackRiga) [reply Assets/_Scripts/1Data/Scriptables/BuildingType.cs:15]: Can't be named the same thing as the class. Feel free to rename the class to make this possible though

File: Assets/_Scripts/3Managers/BaseBuildingValidator.cs (hunk 2 above)
Comment (JackRiga) [reply Assets/_Scripts/3Managers/BaseBuildingValidator.cs:44]: Skip for now

File: Assets/_Scripts/3Managers/BaseBuildingValidator.cs (hunk 3 above)
Comment (JackRiga) [reply Assets/_Scripts/3Managers/BaseBuildingValidator.cs]: Agreed

**Review by JackRiga — APPROVED** [review]

**Pull request merged**